tests/test_pandablocks.py

```diff
@@ -128,11 +465,11 @@ def test_save():
     )
     response_bytes = "\n".join(STATE_RESPONSES).encode() + b"\n"
     assert (
-        conn.receive_bytes(response_bytes[:109])
+        conn.receive_bytes(response_bytes[:107])
```

Comment:
This test is a bit broken - the original index 109 ends up chopping a word in half: `\n.\n!ta <break> bledata\n.`

I think the index should be 107, as that neatly aligns with the punctuation in `response_bytes`, but weirdly the test passes with a whole range of indexes around this number if both numbers are the same. It strikes me as odd that the test can pass when there are partial strings being sent.